buildings_tools/src/buildings_tools/DrawBuildingAction.java

```diff
@@ -279,13 +296,20 @@ private void drawingStart(MouseEvent e) {
         drawStartPos = mousePos;
 
         Node n = MainApplication.getMap().mapView.getNearestNode(mousePos, OsmPrimitive::isUsable);
-        if (n == null) {
-            building.setBase(latlon2eastNorth(MainApplication.getMap().mapView.getLatLon(mousePos.x, mousePos.y)));
+
+        if (ToolSettings.Shape.RECTANGLE.equals(ToolSettings.getShape())) {
+            if (n == null) {
+                building.setBase(latlon2eastNorth(MainApplication.getMap().mapView.getLatLon(mousePos.x, mousePos.y)));
+            } else {
+                building.setBase(n);
+            }
+            mode = Mode.Drawing;
+            updateStatusLine();
         } else {
-            building.setBase(n);
+            building.setBase(latlon2eastNorth(MainApplication.getMap().mapView.getLatLon(mousePos.x, mousePos.y)));
+            mode = Mode.Drawing;
+            updateStatusLine();
```

Comment:
Couldn't the two lines above be outside the `if` and not be duplicated?

Also another spot where I feel like the default Rectangle should be in the `else`